ratelimit/registry.go

```diff
@@ -55,13 +64,18 @@ func NewSwarmRegistry(swarm Swarmer, settings ...Settings) *Registry {
 	return r
 }
 
+// Close teardown Registry and dependent resources
+func (r *Registry) Close() {
+	close(r.quit)
+}
+
 func (r *Registry) get(s Settings) *Ratelimit {
 	r.Lock()
 	defer r.Unlock()
 
 	rl, ok := r.lookup[s]
 	if !ok {
-		rl = newRatelimit(s, r.swarm)
+		rl = newRatelimit(s, r.swarm, r.swimOptions, r.redisRing)
```

Comment:
we don't need to pass the r.swimOptions here, it is enough to pass in r.swarm. See the comment in `newClusterRatelimit and `newRatelimit`.